src/query/interpreter.cpp

```diff
@@ -510,6 +513,77 @@ std::optional<std::string> StringPointerToOptional(const std::string *str) {
   return str == nullptr ? std::nullopt : std::make_optional(*str);
 }
 
+CommonStreamInfo GetCommonStreamInfo(StreamQuery *stream_query, ExpressionEvaluator &evaluator) {
+  return {.batch_interval = GetOptionalValue<std::chrono::milliseconds>(stream_query->batch_interval_, evaluator),
+          .batch_size = GetOptionalValue<int64_t>(stream_query->batch_size_, evaluator),
+          .transformation_name = stream_query->transform_name_};
+}
+
+std::vector<std::string> EvaluateTopicNames(ExpressionEvaluator &evaluator,
+                                            std::variant<Expression *, std::vector<std::string>> topic_variant) {
+  return std::visit(utils::Overloaded{[&](Expression *expression) {
+                                        auto topic_names = expression->Accept(evaluator);
+                                        MG_ASSERT(topic_names.IsString());
+                                        return utils::Split(topic_names.ValueString(), ",");
+                                      },
+                                      [&](std::vector<std::string> topic_names) { return topic_names; }},
+                    std::move(topic_variant));
+}
+
+Callback::CallbackFunction GetKafkaCreateCallback(StreamQuery *stream_query, ExpressionEvaluator &evaluator,
+                                                  InterpreterContext *interpreter_context,
+                                                  const std::string *username) {
+  constexpr std::string_view kDefaultConsumerGroup = "mg_consumer";
+  std::string consumer_group{stream_query->consumer_group_.empty() ? kDefaultConsumerGroup
+                                                                   : stream_query->consumer_group_};
+
+  auto bootstrap = GetOptionalStringValue(stream_query->bootstrap_servers_, evaluator);
+  if (bootstrap && bootstrap->empty()) {
+    throw SemanticException("Bootstrap servers must not be an empty string!");
+  }
+  auto common_stream_info = GetCommonStreamInfo(stream_query, evaluator);
+
+  return [interpreter_context, stream_name = stream_query->stream_name_,
+          topic_names = EvaluateTopicNames(evaluator, stream_query->topic_names_),
+          consumer_group = std::move(consumer_group), common_stream_info = std::move(common_stream_info),
+          bootstrap_servers = std::move(bootstrap), owner = StringPointerToOptional(username)]() mutable {
+    std::string bootstrap = bootstrap_servers
+                                ? std::move(*bootstrap_servers)
+                                : std::string{interpreter_context->config.default_kafka_bootstrap_servers};
```

Comment:
```suggestion
                                : interpreter_context->config.default_kafka_bootstrap_servers;
```
I know this is not connected to the pr, but it is just a small improvment, because I think it is not necessary to create a temporary, the [ternary operator works](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYgATKVpMGoZAFI9AIQuXSS%2BsgJ4BlRugDCqWgFcWDEJIAHKRuADJ4DJgAcr4ARpjE%2BgCcpAAOqAqEzgyePn4BwemZTgLhkTEs8Yl6KfaYjtlCBEzEBLm%2B/kF2mA4lDE0tBGXRcQnJds2t7fldCpNDESOVYzUAlHao3sTI7BxmGgCCFgDMEcg%2BWADUZsfuznPEmKw32PtHeqcM595XN%2B4PEWALzeJzOF0w11u3gYJCwj3QAH0WExUsDDqCvuDIe5vE5aIQAJ5oo6HB7eRyXITXADs1kOlwZlIgEQIlzwq0uIDZZlpeB5ABEaZZLnN0CAUJtWX8/tc9HpHCRZXpsTK8CrbiKCGKQG5aDdhQK3oymWgGHNKRYAGwijlcvm0hQAOntgp5wtF4rQuPV7iVaFSBMuCuISp9bLDHp1DHQeuOBup/KNjKEECEVqtNs53IdzoFQqTxoZTrVN0FAFouPqC4XI16pbcZRY9CxUAA3CHB0PSjUlhsayO6qv0xmG4cMgB%2BKY5bs12rrYab6CIIabYd77hlA%2BjsfjibHbME3LjIITQ5JBxZl2REWZh5awGQpCDCBaACpX5d763p7Tq1SFJWtKAYmx77v%2BehuhBp6gQcxputWxq1pKC5yicaEfJcngBoCB6XFQYhKJcsTEMYyAIGYACs7gME2Z6FsmIqQoK97IExpaXIEkIAGKzuKLbthAAG2iKEEwXBp77kh3rdr6tEfOhxyXDCrIsgkmBzIClHUbRYkMQoxxugZ0F0rBemSG65nGdW8H7ohWqeshMlKgpJyXAAsm2OERJcBDEN4ELEaR5FUTRqG6fR/5MZ%2BxAPpcYBgOxnE3Dxkb8ZggnHMJCjmeFNJ7qZRb2RK0l9rJqHyRVinKQeBBqRpJhaaFokmQKHDrLQnAUbw/gcFopCoJwG5WDYIqbNsEInHovAEJobXrAA1iAxyBI61KWpaGjHBtSRBJtcqGJwkjdbN/WcLwCggBopAzb1bWkHAsBIGgLCpHQCTkJQz2vfQiTnMYwBcHoGhcHwdC1cQF0QLEJ2xBELQEpwPCkM9bCCAA8gwtAI7dpBYMiJjiDj%2BCPA07YXTjmCqPUuK7H1qkdTj%2BKBcQBKeFgJ2%2BXgLCI3dVBGMACgAGp4JgADuaOpIwPMyIIIhiOwUgy/IShqCdugGEYJgoDYNiGHgsQXZA6yoKkfTk2WoqluYw1WFw1KXGWaPHOdPT1H0rjRtM/gg2EiwVFUEhpBkWQCF7gdFCHDDDP7Ywg3UDQCAMUxeB0gfx30ScLOUoyJHH8xh3ngzRznEjrAoY07Po12PLsPDtZ1x04wNHCqIElplpakhBprwCXIDjoaI6XCXBAQ3WFYT64IQiqTU%2BngvW9K4fHoqzTbNqzrAgTxwpQC36NSjpJBoGiWscGgUdSXBt8ckgURrh28NzXDH6QPV9c352Xdd6/3TAiAgJKVIuIPoQC%2BovKumB8DLirvwWWohxCK1gcrFQ6gca%2BUwLXXgosSKpB5vXDgXVX4nWbmjXEQDWSoCoJcVu7dO7d3%2Bn3PQA8h4j3nt9BIspjgrzXrdDepBFrHGOI6QRIjRFiIOhwI6RCm5nTsF/G6Wg%2BEMymtI9%2BsiFFzVIO2CG2QAhAA%3D) well with move/copy branches.

P.S.: Maybe I wrote the original code, in that case sorry for not noticing when I wrote it.

Reply:
I wrote it so no worries. :smile: 
This is the case:
https://godbolt.org/z/9sMErdhj3

The config is `const` so it forces const references for both branches in the ternary, disabling the move. clang-tidy caught that which is extra nice!

Reply:
Nice!